File: JavaScript/src/Internals/Functions/String/Function_LEN.js
import { Function_1 } from '../Function_1.js';
import { Operand } from '../../../Operand.js';
import { StringCache } from '../../../Internals/StringCache.js';

class Function_LEN extends Function_1 {
    constructor(func1) {
        super(func1);
    }

    Evaluate(engine, tempParameter) {
        let args1 = this.func1.Evaluate(engine, tempParameter);
        if (args1.IsNotText) {
            args1 = args1.ToText('Function {0} parameter is error!', 'Len');
            if (args1.IsError) {
                return args1;
            }
        }
        return Operand.Create(args1.TextValue.length);
    }

    toString(stringBuilder, addBrackets) {
        this.AddFunction(stringBuilder, 'Len');
    }
}

export { Function_LEN };

File: java/src/main/java/toolgood/algorithm/internals/functions/string/Function_LEN.java
package toolgood.algorithm.internals.functions.string;

import toolgood.algorithm.internals.functions.FunctionBase;
import toolgood.algorithm.Operand;
import toolgood.algorithm.AlgorithmEngine;
import toolgood.algorithm.internals.functions.Function_1;

public class Function_LEN extends Function_1 {
    public Function_LEN(FunctionBase func1) {
        super(func1);
    }

    @Override
    public Operand Evaluate(AlgorithmEngine work, java.util.function.BiFunction<AlgorithmEngine, String, Operand> tempParameter) {
        Operand args1 = func1.Evaluate(work, tempParameter);
        if (args1.IsNotText()) {
            args1 = args1.ToText("Function '{0}' parameter is error!", "Len");
            if (args1.IsError()) {
                return args1;
            }
        }
        return Operand.Create(args1.getTextValue().length());
    }

    @Override
    public void toString(StringBuilder stringBuilder, boolean addBrackets) {
        AddFunction(stringBuilder, "Len");
    }
}
